### PR TITLE
MGMT-15520: Appliance deployment bootable ISO

### PR DESCRIPTION
**Pull request by danielerez**
Added a command for building a deployment ISO that handles disk image resize and clone into a target device.

The ISO can be used as an alternative for manually cloning the appliance disk image with a 'dd' or 'virt-resize'.
I.e. post boot, the ISO automatically invokes 'virt-resize' command that clones the included appliance.raw file into a specified target device.

For more details, see 'Deployment ISO' section in the user-guide.

### CODE REVIEW

File: pkg/asset/deploy/deploy_iso.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	}
+
+	return nil
+}
+
+func (i *DeployISO) buildDeploymentIso(envConfig *config.EnvConfig, applianceConfig *config.ApplianceConfig) error {
+	if fileName := envConfig.FindInAssets(consts.ApplianceFileName); fileName == "" {
+		logrus.Infof("The appliance.raw disk image file is missing.")
+		logrus.Infof("Run 'build' command for building the appliance disk image.")
+		logrus.Exit(0)
```
Comment (nmagnezi) [inline pkg/asset/deploy/deploy_iso.go:84]: try to see what happens if you set Exit(1), since it does fail to create the ISO in this case.

**Review by nmagnezi — COMMENTED** [review]
Great work!
very minor comments

File: pkg/asset/deploy/deploy_iso.go (hunk 1 above)
Comment (danielerez) [reply pkg/asset/deploy/deploy_iso.go:84]: Done

File: data/scripts/bin/deploy.sh.template [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+) &
+
+printf '\\e{yellow}Preparing to start appliance disk image cloning...\\e{reset}\n' | tee $prepare_issue
+
+# Load appliance image
+podman load -q -i /run/media/iso/deploy/{{.ApplianceImageTar}}
+
+# Create a loop device for each appliance part
+APPLIANCE_FILES="/run/media/iso/deploy/{{.ApplianceFileName}}*"
+loop_index=2 # loop0 and loop1 are already used by CoreOS
```
[the remark this answers is not in the document]
Comment (danielerez) [reply data/scripts/bin/deploy.sh.template:27]: Good point, changed the flow to not use indices.

File: data/scripts/bin/deploy.sh.template [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  device="/dev/loop$loop_index"
+  losetup $device $f
+  loop_sizes[$loop_index]=`blockdev --getsz $device`
+  ((loop_index+=1))
+done
+
+# Create a device map using the loop devices
+(
+  start=0
+  for (( i=2; i<loop_index; i++ ))
```
[the remark this answers is not in the document]
Comment (danielerez) [reply data/scripts/bin/deploy.sh.template:41]: done

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Pull request merged**